apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -0,0 +1,313 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Data;
+using System.Data.Common;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Apis.Auth.OAuth2;
+using Google.Cloud.Spanner.V1;
+
+// ReSharper disable UnusedParameter.Local
+namespace Google.Cloud.Spanner
+{
+    /// <summary>
+    /// A SpannerConnection represents a connection to a single Spanner database.
```

Comment:
A single grpc channel can support multiple 'sessions' to spanner.
A SpannerConnection is an ADO.Net-only concept and is similar to a session, but it will actually use multiple sessions internally if required.